server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala

```diff
@@ -14,10 +14,25 @@ import scodec.bits._
 object RequestLogger {
   private[this] val logger = getLogger
 
-  def apply(logHeaders: Boolean, logBody: Boolean)(service: HttpService)(implicit strategy: Strategy): HttpService = Service.lift{ req =>
-    Stream(req)
-      .through(Logger.messageLogPipe[Request](logHeaders, logBody)(logger))
-      .evalMap[Task, Task, MaybeResponse]{req => service(req)}
-      .runFoldMap(identity)
+  def apply(logHeaders: Boolean, logBody: Boolean)(service: HttpService)(implicit strategy: Strategy): HttpService =
+
+    Service.lift{ req =>
+      if (!logBody) {
+        Logger.logMessage(req)(logHeaders, logBody)(logger)(strategy) >> service(req)
+      } else {
+        async.unboundedQueue[Task, Byte].flatMap { queue =>
+
+          val newBody = Stream.eval(queue.size.get)
+            .flatMap(size => queue.dequeue.take(size.toLong))
```

Comment:
I honestly went with `Queue` because it seemed as though it was meant for it. However since technically the loading and unloading happens concurrently just on another thread I think we would be safe with a StringBuilder.